src/data_race.rs

```diff
@@ -441,21 +441,33 @@ impl MemoryCellClocks {
 /// Evaluation context extensions.
 impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for MiriEvalContext<'mir, 'tcx> {}
 pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriEvalContextExt<'mir, 'tcx> {
-    /// Atomic variant of read_scalar_at_offset.
-    fn read_scalar_at_offset_atomic(
+    /// Calculates the MPlaceTy given the offset and layout of an access on an operand
+    fn offset_and_layout_to_place(
```

Comment:
I just realized that this is public (part of a `pub trait`).

In that case I think this should be in `helpers.rs`, and the name should contain "deref". Maybe `deref_operand_and_offset`? (The "layout" part is also implicit in `place.offset` so seems fine to not mention it here either.)